cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AuditLogWriterModule.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.app.guice;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+import io.cdap.cdap.api.auditlogging.AuditLogWriter;
+import io.cdap.cdap.common.guice.NoOpAuditLogModule;
+import io.cdap.cdap.common.runtime.RuntimeModule;
+import io.cdap.cdap.security.auth.MessagingAuditLogWriter;
+
+public class AuditLogWriterModule extends RuntimeModule {
+
+  /**
+   * Guice modules for In Memory use case of AuditLogWriter.
+   * Returns No Op for now. This may change in the future.
+   */
+  @Override
+  public Module getInMemoryModules() {
+    return new NoOpAuditLogModule();
+  }
+
+  /**
+   * Guice modules for Standalone use case of AuditLogWriter.
+   * Returns No Op for now. This may change in the future.
+   */
+  @Override
+  public Module getStandaloneModules() {
+    return new NoOpAuditLogModule();
+  }
+
+  /**
+   * Guice modules for Distributed use case where the audit events would be written to a messaging queue like tms.
+   */
+  @Override
+  public Module getDistributedModules() {
+    return new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(AuditLogWriter.class).to(MessagingAuditLogWriter.class).in(Scopes.SINGLETON);
```

Comment:
Consider returning the NoOp one if logging is disabled. This would ensure we don't spend cycles on all the machinery and is more safe from bugs perspective.

Reply:
But please log if it's enabled or disabled